### DISCUSSION TITLE
Getting the current index of a RepeatableEntry in Infolist

### DISCUSSION THREAD

**Discussion by sherifiskander** [filamentphp/filament#9081] [General]
Hello,

Got a repeatable entry with some data passed into it. It works fine, just need to get the current index of the loop to use for array matching to get some other data.

Tried accessing $data, but it comes out empty.

**Comment by leandrocfe** [comment]
You could create an [accessor](https://laravel.com/docs/10.x/eloquent-mutators#defining-an-accessor)
```php
protected function customItems(): Attribute
{
    return Attribute::make(
        get: function () {
            return Arr::map($this->your_column, function ($item, $index) {
                $item['index'] = $index + 1;
                return $item;
            });
        },
    );
}
```

```php
RepeatableEntry::make('custom_items')
    ->schema([
        TextEntry::make('index')
...
```

**Reply by sherifiskander** [reply]
Thanks! I already did something similar. How would I for example set the label to index number such as? 

textentry::make('index')->label(iteration number)

**Reply by leandrocfe** [reply]
correct, using `label`

**Reply by sherifiskander** [reply]
Yes. But where do I get iteration number from?

**Reply by leandrocfe** [reply]
Hum, understood. I think you can't do it...
